enhancements/network/ovn-secondary-networks.md

```diff
@@ -0,0 +1,240 @@
+
+### Goals
+- create flat layer 2 overlays
+- connect a newly created pod to these overlay networks
+- these overlays can optionally be a subnet
+- IPv6 support is welcome (not an immediate priority)
+- optional connection to the external networks
+
+### Non-Goals
+- network policy concerns. There isn't a requirement for having network
+  policies for the secondary networks. There is an
+  [enhancement proposal](enhancements/network/multi-networkpolicy.md) about
+  this subject, not scoped around OVN.
+
+## Proposal
+
+### User Stories
+
+#### Isolated tenant network
+A customer can specify a layer 2 overlay network with a subnet via a
+`network-attachment-definition`, which is provisioned without the intervention
+of a system administrator. You can refer to the
+[network attachment definition section](#net-attach-def-samples) for examples.
+
+Afterwards, the users can then request their pods - and / VMs - to connect to
+those overlays as they would for any other secondary network.
+
+### Implementation Details/Notes/Constraints
+A new CRD will be introduced, named `OVNSecondaryNetwork`. When this CRD is
+provisioned, a `NetworkAttachmentDefinition` object will be rendered.
+
+The new CRD will look like:
+
+```golang
+type OVNSecondaryNetwork struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    Spec OVNSecondaryNetworkSpec `json:"spec,omitempty"`
+}
+
+type OVNSecondaryNetworkSpec struct {
+    // Subnet is a RFC 4632/4291-style string that represents an IP address and prefix length in CIDR notation
+    Subnet                  string `json:"subnet"`
+    HasExternalConnectivity bool   `json:"hasExternalConnectivity,omitempty"`
+}
+```
+
+A controller will watch out for the creation / deletion of these
+`OVNSecondaryNetwork`s, and will render and provision the corresponding
+`NetworkAttachmentDefinition`s.
+
+#### Network-attachment-definition provisioning
+When a new network attachment is provisioned (or removed), the corresponding
+controller will create a logical switch (named
+`secondary_<namespace>_<networkName>`), which will assign IP addresses in the
+configured range (defined by the `OVNSecondaryNetwork`).
+
+A removed `Network-Attachment-definition` with present attachment will
+trigger a reconciliation loop, which will cause the net-attach-def entity to be
+reprovisioned; if the network attachment definition is removed and the
+corresponding OVN logical switch does not have any logical ports attached
+triggers the logical switch to be deleted.
+
+#### Adding / removing a pod using an OVN secondary network
+Adding / removing a pod will lead to a logical switch port to be created,
+attached to the corresponding logical switch.
+
+A static MAC / IP address can be requested for the pods by specifying those in
+the annotations field. The requested IP address **must** be in range of the
+corresponding `OVNSecondaryNetwork`.
```

Comment:
What happens if it's not? Does only the secondary attachment fail? Does the entire pod setup fail because OVN is handling both of them and can do that? Does some event get posted to the pod if the whole pod's setup doesn't fail?